spec/mongoid/association/embedded/embedded_in/proxy_spec.rb

```diff
@@ -616,4 +616,31 @@
       end
     end
   end
+
+  context "when assigning a hash" do
+    let(:building_address) { BuildingAddress.new }
+
+    before do
+      building_address.building = { name: "Chrysler" }
+    end
+
+    it "creates the objects correctly" do
+      expect(building_address.building).to be_a(Building)
+      expect(building_address.building.name).to eq("Chrysler")
+    end
+  end
+
+  context "when assigning an association with a hash" do
```

Comment:
I am not sure if the context description reflects its content. The difference between the test above and this one is that above we assign hash once, and here twice. I do not see that we assign "an association with a hash" here.

Reply:
ah right this should replacing

```suggestion
  context "when replacing an association with a hash" do
```